server/service/gov/kie/kie_distributor.go

```diff
@@ -0,0 +1,202 @@
+		ID:     id,
+		Value:  string(yamlByte),
+		Status: p.Status,
+	}
+	_, err = d.client.Put(context.TODO(), kv, kie.WithProject(project))
+	if err != nil {
+		return err
+	}
+	d.lbPolicies[p.GovernancePolicy.Name] = p
+	return nil
+}
+
+func (d *Distributor) Delete(id, project string) error {
+	err := d.client.Delete(context.TODO(), id, kie.WithProject(project))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Distributor) List(kind, project, app, env string) ([]byte, error) {
+	list, _, err := d.client.List(context.TODO(),
+		kie.WithKey("beginWith("+PREFIX+toSnake(kind)+")"),
+		kie.WithLabels(map[string]string{AppKey: app, EnvironmentKey: env}),
+		kie.WithRevision(0),
+		kie.WithGetProject(project))
+	if err != nil {
+		return nil, err
+	}
+	var r []*gov.LoadBalancer
+	for _, item := range list.Data {
+		goc := &gov.LoadBalancer{
+			GovernancePolicy: &gov.GovernancePolicy{},
+		}
+		spec := make(map[string]interface{})
+		specJSON, _ := yaml.YAMLToJSON([]byte(item.Value))
+		err = json.Unmarshal(specJSON, &spec)
+		if err != nil {
+			return nil, err
+		}
+		goc.ID = item.ID
+		goc.Status = item.Status
+		goc.Name = item.Key
+		goc.Spec = spec
+		goc.Selector.App = item.Labels[AppKey]
+		goc.Selector.Environment = item.Labels[EnvironmentKey]
+		goc.CreatTime = item.CreatTime
+		goc.UpdateTime = item.UpdateTime
+		r = append(r, goc)
+	}
+	b, _ := json.MarshalIndent(r, "", "  ")
+	return b, nil
+}
+
+func (d *Distributor) Get(id, project string) ([]byte, error) {
+	kv, err := d.client.Get(context.TODO(), id, kie.WithGetProject(project))
+	if err != nil {
+		return nil, err
+	}
+	goc := &gov.LoadBalancer{
+		GovernancePolicy: &gov.GovernancePolicy{},
+	}
+	goc.ID = kv.ID
+	goc.Status = kv.Status
+	goc.Name = kv.Key
+	goc.Spec = kv
+	goc.Selector.App = kv.Labels[AppKey]
+	goc.Selector.Environment = kv.Labels[EnvironmentKey]
+	goc.CreatTime = kv.CreatTime
+	goc.UpdateTime = kv.UpdateTime
+	b, _ := json.MarshalIndent(goc, "", "  ")
+	return b, nil
+}
+
+func (d *Distributor) Type() string {
+	return svc.ConfigDistributorKie
+}
+func (d *Distributor) Name() string {
+	return d.name
+}
+
+func initClient(endpoint string) *kie.Client {
+	client, err := kie.NewClient(
+		kie.Config{Endpoint: endpoint,
+			DefaultLabels: map[string]string{},
+		})
+	if err != nil {
+		log.Fatalf("init kie client failed, err: %s", err)
+	}
+	return client
+}
+
+func new(opts config.DistributorOptions) (svc.ConfigDistributor, error) {
+	//ep := config.GetString("gov.kie.endpoint", "")
+	return &Distributor{name: opts.Name, lbPolicies: map[string]*gov.LoadBalancer{}, client: initClient(opts.Endpoint)}, nil
+}
+
+func toSnake(name string) string {
+	if name == "" {
+		return ""
+	}
+	temp := strings.Split(name, "-")
+	var s string
+	for num, v := range temp {
+		vv := []rune(v)
+		if num == 0 {
+			s += string(vv)
+			continue
+		}
+		if len(vv) > 0 {
+			if vv[0] >= 'a' && vv[0] <= 'z' { //首字母大写
+				vv[0] -= 32
+			}
+			s += string(vv)
```

Comment:
Use StringBuffer or StringBuilder for much better performance

Reply:
done